### ISSUE TITLE
[1.20.1] Strange problem with IronJetpacks and KubeJS

### ISSUE THREAD

**Issue by lays24mc** [KubeJS-Mods/KubeJS#709]
### Minecraft Version

1.20.1

### KubeJS Version

2001.6.3-build.47

### Rhino Version

2001.2.2-build.6

### Architectury Version

9.1.12-forge

### Forge/Fabric Version

Forge-47.1.3

### Describe your issue

First, I apologize for my bad english, but I hope you understand me anyway.

**Describe the bug**

The bug is that as soon as I insert a custom script in KubeJS, even if it has nothing to do with IronJetpacks, all jetpacks disappear from the game. As example, with a simple script that change the adorn crate (This should actually be an Emerald Jetpacks, which I own):

![grafik](https://github.com/KubeJS-Mods/KubeJS/assets/38819519/a760e932-e337-4dad-9317-3d3ed02a9e59)

The interesting thing is that it only happens when I restart the game. If I remove the scripts while the game is active, use the /reload command, the jetpacks are there again and remain even if I insert the scripts again during the game.  It feels like KubeJS removes all Jetpack recipes the first time you launch the modpack. Unfortunately, I can't find any evidence of this in the logs. I therefore also still know if it is a bug from KubeJS or from Ironjetpacks.

Here is an example script: 

```
ServerEvents.recipes(ev => { 

    ev.remove({ output: 'adorn:crate' })
    ev.shaped(
        Item.of('adorn:crate', 4), // output
        [
            'BAB',
            'B B',
            'BAB'
        ],
        {
            A:'#forge:rods/wooden',
            B:'#minecraft:logs'
        }
    ).id('mwf:adorn_crate')
})
```
 
- Cucumber: 1.20.1-7.0.1
- Iron Jetpacks: 1.20.1-7.0.1

**Comment by ukmcplyr** [comment]
I'm seeing this bug in 1.19.2 as well. Similar to the OP I can reproduce with just the following server script:

```js
ServerEvents.recipes(event =>
{
  event.shapeless('minecraft:gravel', ['3x minecraft:flint'])
  event.shapeless('minecraft:wheat_seeds', ['minecraft:wheat'])
})
```

I'm working on slimming down my reproducer instance so can post logs, specific mod versions etc if that'd be helpful.

**Comment by ukmcplyr** [comment]
Yeah, no dice sadly.

Is it worth taking this to Iron Jetpacks' tracker? I'm guessing not...

**Comment by lays24mc** [comment]
Thanks for your help, but unfortunately it didn't help me either. However, is not bad, because I will replace the mod IronJetpacks. The mod seems to make problems with some other mods like Entity Model Feature also :)

If I find time today, I'll pass it along to IronJetpacks if @ukmcplyr  hasn't done it to date. I mean, ask nicely costs nothing :)

I will close it then once here

**Comment by MaxNeedsSnacks** [comment]
> Iron jetpacks and its hardcoded recipes 😢 You could try this startup script:
> 
> ```js
> StartupScripts.recipeSchemaRegistry(event => {
>   event.namespace('ironjetpacks').special('crafting_jetpack_upgrade')
> })
> ```
> 
> But im not sure if that's going to help

fwiw i'm late but this should probably be `StartupEvents`, not `StartupScripts`. If you'd like to give that a try that'd be greatly appreciated, but obviously no rush or pressure ^^ In the meantime, I'll check out if there's anything we can do to support their code recipes

**Comment by ukmcplyr** [comment]
> fwiw i'm late but this should probably be `StartupEvents`, not `StartupScripts`. If you'd like to give that a try that'd be greatly appreciated, but obviously no rush or pressure ^^ In the meantime, I'll check out if there's anything we can do to support their code recipes

Forgot to mention that I did `s/Scripts/Events/` before running the script (the second post-startup error time :D) and `.special()` logged what it was attempting.

If you're able to work sometihng out and need testers, please give me a shout :beer:

**Comment by KuryKat** [comment]
I was having a similar problem with the mod, the Jetpacks were not loading at all on the latest 1.19.2 release which has backported code from 1.20

After speaking to Blake (the developer) about it, he told me that it could be because of the way that IronJetpacks uses the RecipeManager in the latest releases, and some mod was messing with it (the RecipeManager), he told me it could be either a Create Addon (notably they had known issues with Slice 'n Dice) but I believe it could be an issue with KubeJS too since it also does things to the RecipeManager afaik.

The fix for me on 1.19.2 was to downgrade one version, from 6.0.5 to 6.0.4 and all jetpacks loaded correctly.
Maybe, not sure, this could work for 1.20 too, but it would require knowing which version said features (the usage of RecipeManager) were introduced

Hope it helps! 🙌💞

**Comment by ukmcplyr** [comment]
Interesting. I could've sworn I'd tried downgrading Iron Jetpacks a few days ago and it made no difference, but dropping from 6.0.5 to 6.0.4 as above does indeed work.

@KuryKat Thanks a heap! :beers:

**Comment by lays24mc** [comment]
@ukmcplyr if I read it correctly at IronJetpack, the problem should be fixed with the newest (from today) Cucumber version for 1.19.-.4. Whether 1.20.1 still follows, remains to be seen :). Link: https://github.com/BlakeBr0/IronJetpacks/issues/78

**Comment by ukmcplyr** [comment]
Cheers or the heads up @lays24mc! Fingers crossed it'll get resolved for 1.20.x as well :smile:
